jujugui/static/gui/src/app/components/credential-add-edit/credential-add-edit.js

```diff
@@ -62,14 +61,15 @@ class CredentialAddEdit extends React.Component {
     @returns {Object} React component for DeploymentCloud.
   */
   _generateChooseCloud() {
+    const cloudFacade = this.props.cloudFacade;
     return (
       <DeploymentCloud
         acl={this.props.acl}
         addNotification={this.props.addNotification}
         cloud={this.state.cloud}
-        controllerIsReady={this.props.controllerIsReady}
+        controllerIsReady={() => true}
         key="deployment-cloud"
-        listClouds={this.props.controllerAPI.listClouds}
+        listClouds={cloudFacade.clouds.bind(cloudFacade)}
```

Comment:
This is kind of a special case as this component has not yet been converted as it's also used in the deployment flow. This will no longer be required once it's converted to use the cloudFacade.

But lets see if this pattern emerges as more conversions happen, that might be a good idea.

Reply:
As discussed, I changed my mind, it's definitely a good idea :D